engineio/src/asynchronous/async_transports/websocket_general.rs

```diff
@@ -80,37 +73,22 @@ impl AsyncWebsocketGeneralTransport {
         Ok(())
     }
 
-    pub(crate) fn stream(&self) -> WsStream<'_> {
-        WsStream { inner: self }
-    }
-
-    pub(crate) async fn poll(&self) -> Result<Bytes> {
-        let mut receiver = self.receiver.lock().await;
-
-        let message = receiver.next().await.ok_or(Error::IncompletePacket())??;
-        if message.is_binary() {
-            let data = message.into_data();
-            let mut msg = BytesMut::with_capacity(data.len() + 1);
-            msg.put_u8(PacketId::Message as u8);
-            msg.put(data.as_ref());
-
-            Ok(msg.freeze())
-        } else {
-            Ok(Bytes::from(message.into_data()))
-        }
-    }
-}
-
-impl Stream for WsStream<'_> {
-    type Item = Result<Bytes>;
-
-    fn poll_next(
-        self: Pin<&mut Self>,
-        cx: &mut std::task::Context<'_>,
-    ) -> std::task::Poll<Option<Self::Item>> {
-        match ready!(Pin::new(&mut Box::pin(self.inner.poll())).poll(cx)) {
-            Ok(val) => Poll::Ready(Some(Ok(val))),
-            Err(err) => Poll::Ready(Some(Err(err))),
-        }
+    pub(crate) fn stream(&self) -> impl Stream<Item = Result<Bytes>> + '_ {
+        self.receiver
+            .lock()
+            .into_stream()
```

Comment:
nit: does this into_stream consume self.receiver?

Reply:
I don't think so, as far as I understand this consumes the future, otherwise tests like this: https://github.com/1c3t3a/rust-socketio/blob/db405b1661c4be7d779868b3374914a2e3740e47/engineio/src/asynchronous/async_transports/websocket.rs#L144-L155
wouldn't work as the two calls to `stream` each resolve in a different stream.

Reply:
Or am I wrong about this?

Reply:
Maybe a failing of the naming of the other crate. If it works, it works.